test/docs/debug.test.js

```diff
@@ -67,4 +71,20 @@ describe('debug: shell', function() {
     assert.equal(true, lastLog.includes('ISODate'));
   });
 
+  it('should allow to set the `debug` option on a per-connection basis (gh-12700)', async function() {
+    // Disable global debug
+    mongoose.set('debug', false);
+    // `conn1` with active debug
+    const conn1 = mongoose.createConnection(start.uri);
+    conn1.set('debug', true);
+    const testModel1 = conn1.model('Test', testSchema);
+    await testModel1.create({ dob: new Date(), title: 'Connection 1' });
```

Comment:
We usually use the issue number, so in this case `gh-12700`, to create the collection name but `date` is a nifty way of doing it too.